.circleci/config.yml

```diff
@@ -2,65 +2,271 @@ version: 2.1
+          key: pip-packages-v5-{{ .Branch }}-{{ checksum "Pipfile.lock" }}
+          paths:
+            - ~/.local
+            - ~/.cache
+      - run:
+          name: create data model
+          command: |
+            make resetdb
+
+      - run:
+          name: typecheck server
+          command: |
+            make typecheck-server
+
+      - run:
+          name: format server
+          command: |
+            make format-server
+            git diff-index --quiet HEAD -- || (echo "Found unexpected changes!" && git diff && exit 1)
+      - run:
+          name: lint server
+          command: |
+            make lint-server
+
+  build-and-test-server:
+    docker:
+      - image: circleci/python:3.7.3-node-browsers
+        environment:
+          DATABASE_URL: postgresql://root@localhost:5432/arlo-test
+          FLASK_ENV: test
+      - image: circleci/postgres:9.6.3-alpine-ram
+        environment:
+          POSTGRES_USER: root
+          POSTGRES_DB: arlo-test
+    resource_class: xlarge
+    steps:
+      - checkout
+      - setup_remote_docker:
+          docker_layer_caching: true
+      - run: sudo chown -R circleci:circleci /usr/local/bin
+      - run: sudo chown -R circleci:circleci /usr/local/lib/python3.7/site-packages
+      - restore_cache:
+          name: Restore Pip Cache
+          keys:
+            - pip-packages-v5-{{ .Branch }}-{{ checksum "Pipfile.lock" }}
+            - pip-packages-v5-{{ .Branch }}-
+            - pip-packages-v5-
+      - run:
+          name: install
+          command: |
+            pipenv install --dev
+
+      - save_cache:
+          name: Save Pip Cache
+          key: pip-packages-v5-{{ .Branch }}-{{ checksum "Pipfile.lock" }}
+          paths:
+            - ~/.local
+            - ~/.cache
+      - run:
+          name: create data model
+          command: |
+            make resetdb
+      - run:
+          name: test-server # server tests
+          command: |
+            pipenv run pytest --cov=.
+
+  build-and-test-client:
     docker:
       - image: circleci/python:3.7.3-node-browsers
         environment:
           DATABASE_URL: postgresql://root@localhost:5432/arlo-test
           FLASK_ENV: test
-      - image: circleci/postgres:9.6.2-alpine
+      - image: circleci/postgres:9.6.3-alpine-ram
         environment:
           POSTGRES_USER: root
           POSTGRES_DB: arlo-test
+    resource_class: xlarge
     steps:
       - checkout
+      - setup_remote_docker:
+          docker_layer_caching: true
+      - run: sudo chown -R circleci:circleci /usr/local/bin
+      - run: sudo chown -R circleci:circleci /usr/local/lib/python3.7/site-packages
+      - restore_cache:
+          name: Restore Yarn Cache
+          keys:
+            - yarn-packages-v3-{{ .Branch }}-{{ checksum "yarn.lock" }}
+      - restore_cache:
+          name: Restore Node Cache
+          keys:
+            - npm-packages-v8-{{ .Branch }}-{{ checksum "package-lock.json" }}
```

Comment:
why don't we use yarn for everything? there's a `yarn.lock` file in `arlo` and in `arlo/client`

Reply:
Fixed this.